src/api/analytics/AnalyticsRequest.js

```diff
@@ -63,10 +63,18 @@ class AnalyticsRequest extends AnalyticsRequestDimensionsMixin(
                 dimension += `:${d.filter}`
             }
 
-            request = request.addDimension(
-                dimension,
-                d.items?.map((item) => item.id)
-            )
+            if (d.repetition?.indexes?.length) {
+                d.repetition.indexes.forEach((index) => {
+                    request = request.addDimension(
+                        dimension.replace(/\./, `[${index}].`)
```

Comment:
I think the dimension's `programStageId` is not going to be prefixed to the `dimensionId` but passed via `programStage: { id: [id] }` object nested in the dimension object.
So this will be changed.

Reply:
True, the`replace` will have to be changed but I suppose the end result will be the same format regardless, e.g. `A03MvHHogjR[1].a3kGcGDCuk6`.

Reply:
Yes, what goes in the analytics request is correct.

Reply:
Actually, we don't need any change, as the dimension is already formatted to `programStageId.dimensionId` before this new code is executed 👍 .